### DISCUSSION TITLE
Why doesn't this layout stretch? And how do I fix it?

### DISCUSSION THREAD

**Discussion by martinellison** [slint-ui/slint#4308] [Q&A]
The following layout does not stretch (when I drag the bottom right corner, the size of the window does not change). Why is this so?

I would like it to stretch, so what changes should I make?

```
import { LineEdit } from "std-widgets.slint";
export component TestWindow inherits Window {
    VerticalLayout {
        spacing: 1rem;
        padding: 1rem;
        test_box := LineEdit {
            height: 2rem;
            width: 20rem;
            text: "";
        }
    }
}
```

Here is the rust code that I run the slint code with:
```
/*! test of the slint GUI engine */
slint::include_modules!();
use color_eyre::eyre::Result;
fn main() -> Result<()> {
	color_eyre::install()?;
	let window = TestWindow::new()?;
	window.run()?;
	Ok(())
}
```

**Comment by ogoffart** [comment]
```
            height: 2rem;
            width: 20rem;
```

You are using fixed size, this means that there is no way to resize.
If you remove these lines, it can be resized.
You can also make it resized by adding `alignment:start`, for example, to the layout properties.

**Comment by martinellison** [comment]
Olivier,

Ok thanks, it is stretchy now.

Regards,
Martin
